src/ray/stats/metric.cc

```diff
@@ -98,6 +98,11 @@ Metric::Metric(const std::string &name,
           ". Metric names can only contain letters, numbers, _, and :. "
           "Metric names cannot start with numbers. Metric name cannot be "
           "empty.");
+  if (name.find(':') != std::string::npos) {
+    RAY_LOG(WARNING) << "Metric name contains a : character, which is no longer allowed. "
```

Comment:
users won't ever see this since it's warning level in cpp logs. need to use error level or (preferably) log it at the Python level in the metric wrapper